docs/source/tutorial/data-source.md

```diff
@@ -3,6 +3,424 @@ title: "2. Hook up your data sources"
+    ...
+  }
+
+  ...
+
+  async bookTrip({ launchId }) {
+    const userId = this.context.user.id;
+    return !!this.store.trips.findOrCreate({ where: { userId, launchId } });
+  }
+
+  async cancelTrip({ launchId }) {
+    const userId = this.context.user.id;
+    return !!this.store.trips.destroy({ where: { userId, launchId } });
+  }
+}
+...
+```
+
+A user selects a particular launch and books a trip. The `userId` and `launchId` values are needed to book the trip successfully. Therefore, the `bookTrip` method accepts a `launchId` via its arguments, obtains the `userId` via the `context` object and invokes the `findOrCreate` method on the `trips` table to book the trip.
+
+The `cancelTrip` method requires a `userId` and `launchId` to delete a trip from the `trips` table successfully. Therefore, the `cancelTrip` method performs almost the same operation as the `bookTrip` method except that it invokes the `destroy` method on the `trips` table and deletes the trip.
+
+<h4 id="get-launches">Get All Launches By User</h4>
+
+We need to get all the launches reserved by a user. This calls for a method, `getLaunchIdsByUser`. Copy the method below and add it to the file.
+
+_src/datasources/user.js_
+
+```js
+...
+class UserAPI extends DataSource {
+  constructor() {
+    ...
+  }
+
+  ...
+
+  async getLaunchIdsByUser() {
+    const userId = this.context.user.id;
+    const found = await this.store.trips.findAll({
+      where: { userId },
+    });
+    return found && found.length
+      ? found.map(l => l.dataValues.launchId).filter(l => !!l)
+      : [];
+  }
+}
+...
+```
+
+Let's analyze the code above.
+
+In the `getLaunchIdsByUser` method, a `userId` is accepted via the `context` object. All the trips booked by a user with a particular `userId` are fetched and stored in the `found` variable. If there are trips found, then an array of launch ids are returned else an empty array is returned.
+
+<h4 id="get-launches">Get Booked Status on a Launch</h4>
+
+We need to add a method that can return the booked status of a launch for a particular user. Copy the method, `isBookedOnLaunch` below and add it to the `UserAPI` class.
+
+_src/datasources/user.js_
+
+```js
+...
+async isBookedOnLaunch({ launchId }) {
+  const userId = this.context.user.id;
+  const found = await this.store.trips.findAll({
+    where: { userId, launchId },
+  });
+  return found && found.length > 0;
+}
+```
+
+In the `isBookedOnLaunch` method, we invoke the `findAll` method of the `trips` table to find out if the particular launch passed to the method via `launchId` has been booked by the logged-in user.
+
+In the various methods that we created and copied to the `UserAPI` class, you must have noticed `this.store.users` and `this.store.trips`. These are two tables from our potential SQLite data store. Let's create the store!
+
+<h4 id="create-the-store">Create the Store</h4>
+
+Create an `utils.js` file inside the `src` directory. Now, copy the code below and add to it.
```

Comment:
Remove all of the code related to creating a store -- we're going to have them fork a copy of the repo that already has this code filled in for them :)

Reply:
I have replaced this with a filler link to the supposed code the readers should fork.